### ISSUE TITLE
The detail of ProDe

### ISSUE THREAD

**Issue by a12541115336** [tntek/source-free-domain-adaptation#16]
First Congrats for accepted by ICLR！Recently, I replicated ProDe on Domainnet126. However, results are far from what the paper suggests (about 5%-10%). Besides, I find that the backbone claimed in the paper which is different in the yaml file. resnet50 in yaml but resnet101 in paper. Could you provide with the checkpoints for ProDe on Domainnet126 dataset?

**Comment by mrmenand** [comment]
Dear Authors,

Thank you for your detailed and patient response, as well as for sharing the original training scripts, logs, and checkpoints. We truly appreciate your efforts to facilitate a faithful comparison.

Based on the newly provided materials, we have carefully re-run the reproduction experiments. While the performance improved slightly, it still falls short of the reported results and is surprisingly lower than the performance of the public GitHub version.

Upon a deeper inspection of the provided code and logs, we noticed several critical discrepancies between the actual implementation and the paper's statements. We would appreciate your clarification on these points, as they appear to be the key factors driving the reported performance:

1. Unreported Learning Rate:
The provided logs explicitly use `lr=1e-3`, whereas the public config specifies `5e-3`. While correcting this does improve performance, it still fails to reach the reported results. 

2. Hardcoded Correction Strength ($\omega=0.1$):
In the newly provided code, the proxy denoising weight is hardcoded as follows:

`new_clip = (outputs_test_new - 0.1 * logtis_bank[tar_idx].cuda()) + clip_score.cuda()`

This implementation effectively sets $\omega=0.1$. However, Appendix D explicitly claims: "On all four datasets, we set $(\omega, \alpha, \beta) = (1, 1, 0.4)$". Furthermore, the parameter sensitivity analysis (Fig. 5) only explores $\omega$ values from $0.5$ to $1.4$. Setting $\omega=0.1$ fundamentally alters the dynamic of the proxy denoising mechanism and contradicts the paper.

3. Severe Inconsistencies in Loss Weights ($\alpha, \beta, \gamma$):
We noticed that the IIC loss weight ($\alpha$) in the newly shared code defaults to `1.0`, whereas the public GitHub configuration for DomainNet sets `IIC_PAR: 1.3`. More importantly, when reviewing the overall loss formulation:

`total_loss = args.alpha * iic_loss + args.beta * pl_loss - args.gama * gent_loss`

We found that the actual hyperparameters used in the public repository vary drastically across datasets:
- Office-31: $\alpha=1.3, \beta=0.1, \gamma=1.0$
- Office-Home: $\alpha=1.3, \beta=0.4, \gamma=1.0$
- Mini-DomainNet: $\alpha=1.3, \beta=0.4, \gamma=0.5$
- VisDA: $\alpha=1.0, \beta=0.05, \gamma=0.1$

This highly dataset-specific tuning directly contradicts the claim in Appendix D that unified parameters were used across all datasets.

We completely understand that domain adaptation tasks can be sensitive to environmental factors. However, the undocumented change in learning rate, the hardcoded $\omega=0.1$, and the dataset-specific loss weights appear to be critical algorithmic interventions rather than mere randomness. We look forward to your insights on how these hidden factors contribute to the final performance.

Thank you again for your time and open academic discussion.

**Comment by Kalaniik** [comment]
Thank you for your detailed feedback. **We highly value open academic discussion**, as rigorous peer scrutiny helps us continuously refine our project. To facilitate our discussion, **we would like to share our perspectives on the details you raised:**
**1. Reproducibility & Source Model Weights**
If you are still unable to fully reproduce the results, **we strongly recommend using the checkpoints we provided above** to exclude variations in reproducibility.
**2. Learning Rate Discrepancy**
The learning rate mismatch was an **unintended oversight during code integration** into the unified framework. **We have pushed a commit to fix this.** All other parameters in the newly provided framework remain consistent with the public project.
**3. The Proxy Denoising Weight ($\omega$ vs. Source Logits Coefficient)**
We respectfully suggest there might be a slight misunderstanding regarding the distinction between the hyperparameter $\omega$ and the source logits coefficient in our implementation. To clarify, the mechanism operates as follows:

`new_clip =  ω  * (outputs_test_new - [CoffS] * logtis_bank[tar_idx].cuda()) + clip_score.cuda()`

Here, **$\omega$ controls the overall denoising strength**, whereas the `[CoffS]` **solely scales the source logits term.** For DomainNet-126, the `[CoffS]` is indeed set to 0.1 in both our public repository (`./src/methods/net/ProDe.py`) and the newly provided code, **while it remains exactly 1.0 for the other three datasets**. Meanwhile, **$\omega$ remains 1.0 across all four datasets, consistent with our paper.**
As is common practice in related works, we empirically found that setting this source coefficient to 0.1 yields better performance due to the specific characteristics of DomainNet-126. Given the subsequent exponential operation, **it acts purely as a scaling constant and does not "fundamentally alter the dynamic" of our mechanism.**
Furthermore, our parameter sensitivity analysis (Fig. 5) was conducted on the Office-Home (A $\rightarrow$ C) task. **Since $\omega$ is exactly 1.0 on this dataset, exploring the [0.5, 1.4] range is entirely consistent and reasonable.**
**4. Inconsistencies in Hyperparameters (α, β, γ)**
You are correct. This was a **documentation oversight** during our paper revision process. We unfortunately failed to synchronize the dataset-specific parameters from our codebase into the manuscript. **We will correct these details in our next arXiv update.** Thank you for your meticulous review.
**5. Concluding Remarks**
**The inconsistency regarding hyperparameters was a typographical error in the text, not a strategic change to the algorithm.** As our code and checkpoints are fully open-sourced, the actual implementation is readily verifiable. **We are updating the description to align with the experimental logs.** We thank you for the feedback, which helps ensure the technical accuracy of this project.

**Comment by mrmenand** [comment]
Dear Authors,

Thank you for your patient and detailed response. Your explanation has clarified most of my previous questions.

Regarding your suggestion in Point 1 to use the provided checkpoints and logs to exclude variations in reproducibility, I took your advice and conducted a careful review of these materials. However, this led me to notice one final, crucial detail regarding the hyperparameter configurations. It appears that the shared logs contain several extra parameters that do not exist in the public GitHub repository.

Specifically, I would appreciate your insights on the following observations:

1. Undocumented Extra Parameters: 
The configuration header in the log file reveals several parameters completely absent from the public codebase, such as `threshold: 0.7`, `p_steps: 6`, `delta_lr: 0.5`, and `ent_par: 1.0`. 

2. The Dynamically Fluctuating `args.w`: 
While the log header displays `w: 1.0` (which aligns with your statement that $\omega$ strictly remains a static 1.0), the actual training iteration outputs show `args.w` as a dynamically changing value that fluctuates drastically. For instance:
- `Iter:470/14100; ... args.w = 0.264...`
- `Iter:3760/14100; ... args.w = 0.103...`
- `Iter:7990/14100; ... args.w = 0.054...`

Since you kindly clarified that the proxy denoising weight strictly operates as a static scaling constant without fundamentally altering the mechanism's dynamic, could you explain what this fluctuating `args.w` represents in your private codebase? 

Does this `args.w` (perhaps combined with the unreleased `threshold: 0.7`) function as a hidden dynamic adaptive weight or filtering mechanism that actively participates in the final loss calculation and gradient backpropagation?

I look forward to your clarification on how these extra parameters operate. Thank you again for supporting such a rigorous and open academic discussion!

Best regards,

**Comment by Kalaniik** [comment]
Thank you for your exceptionally close attention to the intricate details of our project. Based on our team's years of experience in the field of Domain Adaptation, we would like to share a broader perspective regarding your recent observations:
**1. Extraneous Testing Parameters**
Our project is a comprehensive open research framework containing various exploratory baselines. The undocumented parameters you noted are **purely legacy testing arguments and have zero impact on the actual ProDe method.**
**2. The Misinterpretation of `args.w`**
The `args.w` printed in the logs is **completely unrelated to the hyperparameter $\omega$** in our paper. It is simply a legacy debugging variable used to track an intermediate metric during training, not a hidden algorithmic weight.
**3. Academic Scope vs. Engineering Artifacts**
We open-sourced our project to ensure authenticity and transparency. However, **an academic paper is not a technical manual.** It must focus on core methodological innovations rather than exhaustively documenting every line of testing code. **We respectfully suggest that your remaining questions regarding parameters can be easily resolved by directly reading the open-sourced code.**
We value equitable academic dialogue and hope to elevate our discussion beyond basic implementation debugging. The fundamental innovation of ProDe lies in **understanding Domain Adaptation from a dynamic perspective.** We warmly welcome any further scientific discussions centered on this core methodology.
